pkg/client/interfaces.go

```diff
@@ -94,6 +94,9 @@ type SonobuoyKubeAPIClient interface {
 	Name(*unstructured.Unstructured) (string, error)
 	Namespace(*unstructured.Unstructured) (string, error)
 	ResourceVersion(*unstructured.Unstructured) (string, error)
+
+	ListNSResources(ns string, filterResources []string) (map[string][]unstructured.Unstructured, error)
+	//ListClusterResources(filterResources []string) ([]unstructured.Unstructured, error)
```

Comment:
I dont think these are needed; it is all in the discovery code and will be eventually extracted to its own plugin.